### PR TITLE
Allow CWD-relative paths with /script load

### PR DESCRIPTION
**Pull request by LemonBoy**
Replace the usage of printfs with glib functions to build the paths.

### CODE REVIEW

**Comment by ailin-nemui** [comment]
doesn't that break ./ pointing to ~/.irssi/scripts?

File: src/perl/perl-core.c [hunk 1]
```diff
@@ -354,10 +357,11 @@ PERL_SCRIPT_REC *perl_script_find_package(const char *package)
 /* Returns full path for the script */
 char *perl_script_get_path(const char *name)
 {
-	struct stat statbuf;
 	char *file, *path;
 
-	if (g_path_is_absolute(name) || (name[0] == '~' && name[1] == '/')) {
+	if (g_path_is_absolute(name) ||
+	    (name[0] == '~' && name[1] == '/') ||
+	    (name[0] == '.' && name[1] == G_DIR_SEPARATOR)) {
```
Comment (ailin-nemui) [inline src/perl/perl-core.c]: inconsistent usage of such things as G_DIR_SEPARATOR. do we want to introduce this windows madness or not

**Comment by ailin-nemui** [comment]
question if we should map ./ to cwd. it doesn't *seem* very useful to me

**Comment by LemonBoy** [comment]
Different strokes for different folks, I find that mapping `./` to entirely different folders depending on the context to be quite counter-intuitive, after all `./` introduces a cwd-relative path.

File: src/perl/perl-core.c (hunk 1 above)
Comment (dequis) [inline src/perl/perl-core.c]: Apparently all the versions of windows derived from NT support / as a path separator. So, everything newer than 9x/ME. I don't think we support non-cygwin windows anyway.

[x] burninate

**Pull request closed without merging**